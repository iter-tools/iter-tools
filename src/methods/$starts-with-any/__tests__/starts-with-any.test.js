/**
 * @generated-from ./$starts-with-any.test.js
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

/* eslint-disable no-unused-vars,import/no-duplicates */

import { startsWithAny, range } from '../../..';
describe('startsWithAny', () => {
  it('returns true if the iterable starts with any of the given values', () => {
    expect(startsWithAny([0, 1], range(1, 10))).toBe(true);
  });
  it('returns true if the iterable starts with all of the given values', () => {
    expect(startsWithAny([1, 1], range(1, 10))).toBe(true);
  });
  it('returns false if the iterable contains but does not start with any of the given values', () => {
    expect(startsWithAny([1], range(0, 10))).toBe(false);
  });
  it('returns false if the iterable does not contain any of given values', () => {
    expect(startsWithAny([1, 3, 4], [2])).toBe(false);
  });
  it('returns false if the iterable is empty', () => {
    expect(startsWithAny([undefined], [])).toBe(false);
  });
});
