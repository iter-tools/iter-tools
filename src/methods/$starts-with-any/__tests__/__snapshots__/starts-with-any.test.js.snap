// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`startsWithAny when iterable is a string warns 1`] = `
Array [
  Array [
    "For string inputs use startsWithAnySubseq instead of startsWithAny",
  ],
]
`;
